lib/bridge/IrcBridge.js

```diff
@@ -428,6 +428,7 @@ IrcBridge.prototype.getThirdPartyProtocol = function(protocol) {
     var servers = this.getServers();
 
     return Promise.resolve({
+        "bot_user_id": this.getAppServiceUserId(),
```

Comment:
If it's up here, it'll be merged wrongly by the synapse. It wants to live in the `instances` list, so that when synapse merges multiple instances lists across multiple AS programs, they all preserve the right value.